src/lua/uri.lua

```diff
@@ -220,6 +267,57 @@ local function values(...)
     return {...}
 end
 
+local char_to_hex = function(c)
+    return string.format("%%%02X", string.byte(c))
+end
+
+-- Encodes a string into its escaped hexadecimal representation.
+local function escape(buf, opts)
+    if type(buf) ~= "string" then
+        error("Usage: uri.escape(string, [opts])")
+    end
+    local options = opts or RFC3986
+    options.unreserved = options.unreserved or RFC3986.unreserved
+    options.plus = options.plus or RFC3986.plus
+
+    local pattern = unreserved(options.unreserved)
```

Comment:
Double `unreserved()` call? It leads to interpreting pattern symbols as unreserved symbols. Look:

Lua escape:

```yaml
# tarantool> uri = require('uri')
# tarantool> uri.escape('^[]$')
---
- ^[]$
...
```

FFI escape:

```yaml
# tarantool> uri = require('uri')
# tarantool> uri.escape('')
---
- '%5E%5B%5D%24'
...
```

It means:

* The Lua implementation has a problem.
* The tests have a problem.
* Removal of those tests in the FFI implementation hides differences between implementations, so we can miss not only Lua implementation problems, but also FFI implementation ones.